#240 Fix silently failing user import mails by excluding legacy com.sun.mail (#241)
cloudogu/usermgt: CHANGELOG.md

and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]
### Security
- Fixed CVE-2026-41293, CVE-2026-43512 and CVE-2026-43515 in Tomcat by updating from 9.0.117 to 9.0.118.

### Fixed

> - [#240] Fixed a bug where the initial-password mails of the CSV user import were silently never sent, because the legacy `com.sun.mail` jar (pulled in transitively by RESTEasy) broke message serialization of the Jakarta Mail stack.

Reviewer: Since this is a bug fix, I suggest to add which versions of Usermgt are affected. This way users can easily identify whether they are affected as well or not.

Author: Done. I added affected versions to the changelog entry in cca6eafa

> - [#240] Fixed the whole app failing to start after excluding that legacy jar, because RESTEasy's builtin multipart provider reflectively needs `javax.mail.MessagingException` at deployment time; added the API-only `javax.mail:javax.mail-api` jar to supply it without reintroducing the broken mailcap.

Reviewer: I am unsure if this level of detail belongs into a changelog. No user or administrator is supposed to handle things on Java Jav levels when it is actually our business. Anyhow, the reason DOES belong somewhere, but we have to figure out where the best place might be.

For now, I suggest to shorten it a bit like so:

```suggestion
- [#240] Fixed failing application starts by providing a consistent mail Jar without reintroducing the broken mailcap.
```

Reviewer: Maybe this belongs to the developer docs, or the code/POM.xml which handles the mentioned dependencies?

Author: I shortened the changelog entry and added a comment on the pom.xml exclusion.
